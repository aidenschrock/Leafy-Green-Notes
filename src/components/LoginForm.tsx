import { useState } from "react";
import Modal from "@leafygreen-ui/modal";
import TextInput from "@leafygreen-ui/text-input";
import Button from "@leafygreen-ui/button";

interface FuncProps {
  handleSubmit(email, password): void;
  open: boolean;
  setOpen: any;
}

const LoginForm: React.FC<FuncProps> = (props) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  return (
    <Modal data-testid="login-form" open={props.open} setOpen={props.setOpen}>
      <TextInput
        data-testid="email-input"
        className="input"
        type="email"
        label="Email"
        errorMessage="Email address invalid."
        description="Enter your email below"
        placeholder="your.email@example.com"
        onChange={(event) => {
          setEmail(event.target.value);
        }}
      />
      <TextInput
        data-testid="password-input"
        className="input"
        type="password"
        label="Password"
        errorMessage="Incorrect password."
        description="Enter your password below"
        onChange={(event) => {
          setPassword(event.target.value);
        }}
      />
      <Button
        id="submit-button"
        type="submit"
        variant={"primary"}
        onClick={() => props.handleSubmit(email, password)}
      >
        Submit
      </Button>
    </Modal>
  );
};

export default LoginForm;
