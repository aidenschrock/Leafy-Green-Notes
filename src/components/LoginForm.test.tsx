import { render, screen, fireEvent } from '@testing-library/react';
import LoginForm from './LoginForm';
import { shallow } from 'enzyme';
import userEvent from '@testing-library/user-event'


  describe('LoginForm', () => {
    it('should render correctly', () => {
      const component = shallow(<LoginForm  open={true} setOpen={null} handleSubmit={null} />);
      expect(component).toMatchSnapshot();
    });
  });

  test('email type', async () => {
    render(<LoginForm  open={true} setOpen={null} handleSubmit={null}/>)
  const textbox = screen.getByTestId('email-input');
  userEvent.type(textbox, 'test@email.com');
  expect(await textbox).toHaveValue('test@email.com');
  });

  test('password type', async () => {
    render(<LoginForm  open={true} setOpen={null} handleSubmit={null}/>)
  const textbox = screen.getByTestId('password-input');
  userEvent.type(textbox, 'P@ssw3rd');
  expect(await textbox).toHaveValue('P@ssw3rd');
  });

  it('submits when clicked', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<LoginForm open={true} setOpen={null} handleSubmit={mockCallBack}/>);
    const buttonElement  = wrapper.find('#submit-button');
    buttonElement.simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('should send data when form is submitted', () => {
    const handleSubmit = jest.fn();
    render(<LoginForm open={true} setOpen={null} handleSubmit={handleSubmit} />)
    const emailInput = screen.getByTestId('email-input');
    const passwordInput = screen.getByTestId('password-input');

    fireEvent.change(emailInput, {target: {value: "test@email.com"}}) 
    fireEvent.change(passwordInput, {target: {value: "P@ssw3rd"}}) 

    const buttonElement  = screen.getByText(/Submit/i)
 
    fireEvent.click(buttonElement)
    expect(handleSubmit).toHaveBeenCalledWith('test@email.com','P@ssw3rd') 
  
  });
  