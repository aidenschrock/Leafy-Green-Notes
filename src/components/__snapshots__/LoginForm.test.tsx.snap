// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly 1`] = `
<Modal
  data-testid="login-form"
  open={true}
  setOpen={null}
>
  <TextInput
    className="input"
    data-testid="email-input"
    description="Enter your email below"
    errorMessage="Email address invalid."
    label="Email"
    onChange={[Function]}
    placeholder="your.email@example.com"
    type="email"
  />
  <TextInput
    className="input"
    data-testid="password-input"
    description="Enter your password below"
    errorMessage="Incorrect password."
    label="Password"
    onChange={[Function]}
    type="password"
  />
  <Button
    id="submit-button"
    onClick={[Function]}
    type="submit"
    variant="primary"
  >
    Submit
  </Button>
</Modal>
`;
